File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.java
class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        HashSet<Integer> hs = new HashSet<>();
        for(int i : banned) hs.add(i);
        int tot = 0, s = 0;
        for(int i = 1 ;i<=n;i++){
            if(s+i>maxSum) return tot;
            if(!hs.contains(i)){
                tot+=1;
                s+=i;
            }
        }

        return tot;
    }
}

File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.py
class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        l = set(banned)
        tot = 0
        s = 0
        for i in range(1,n+1):
            if s+i>maxSum:return tot
            if i not in l:
                s += i
                tot+=1
        return tot

